src/main/java/com/recurly/v3/Pager.java

```diff
@@ -93,4 +92,48 @@ private void clone(Pager<T> pager) {
     this.more = pager.hasMore();
     this.data = pager.getData();
   }
+
+  private class PagerIterator implements Iterator {
+    public PagerIterator(Object[] data, Pager<T> pager) {
+      this.data = data;
+      this.pager = pager;
+    }
+
+    private int position = 0;
+    private Object[] data;
+    private Pager<T> pager;
+
+    public boolean hasNext() {
+      if (position < this.data.length) {
```

Comment:
Could this whole method be `return position < this.data.length || pager.hasMore()`?